ng2-components/ng2-alfresco-core/src/directives/node-restore.directive.ts

```diff
@@ -0,0 +1,262 @@
+/*!
+ * @license
+ * Copyright 2016 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
+import { MdSnackBar } from '@angular/material';
+import { Router } from '@angular/router';
+import { DeletedNodeEntry, PathInfoEntity } from 'alfresco-js-api';
+import { Observable } from 'rxjs/Rx';
+import { AlfrescoApiService } from '../services/alfresco-api.service';
+import { TranslationService } from '../services/translation.service';
+
+@Directive({
+    selector: '[adf-node-restore]'
+})
+export class NodeRestoreDirective {
+    private restoreProcessStatus;
+
+    @Input('adf-node-restore')
+    selection: DeletedNodeEntry[];
+
+    @Input('restoreLocation') location: string = '';
```

Comment:
https://angular.io/guide/styleguide#avoid-aliasing-inputs-and-outputs

can we remove the aliases?